plugin/pkg/admission/taint/admission.go

```diff
@@ -0,0 +1,280 @@
+	"k8s.io/kubernetes/pkg/apis/rbac/validation"
+	"k8s.io/kubernetes/pkg/client/clientset_generated/internalclientset/typed/rbac/unversioned"
+	"k8s.io/kubernetes/pkg/runtime"
+
+	apierrors "k8s.io/kubernetes/pkg/api/errors"
+	clientset "k8s.io/kubernetes/pkg/client/clientset_generated/internalclientset"
+)
+
+const untrusted string = "tpm.coreos.com/untrusted"
+
+func newRbacClient(client clientset.Interface) validation.AuthorizationRuleResolver {
+	c := &rbacClient{client.Rbac()}
+	return validation.NewDefaultRuleResolver(c, c, c, c)
+}
+
+type rbacClient struct {
+	client unversioned.RbacInterface
+}
+
+func (c *rbacClient) GetClusterRole(ctx api.Context, id string) (*rbac.ClusterRole, error) {
+	return c.client.ClusterRoles().Get(id)
+}
+
+func (c *rbacClient) ListClusterRoleBindings(ctx api.Context, options *api.ListOptions) (*rbac.ClusterRoleBindingList, error) {
+	return c.client.ClusterRoleBindings().List(*options)
+}
+
+func (c *rbacClient) GetRole(ctx api.Context, id string) (*rbac.Role, error) {
+	ns, ok := api.NamespaceFrom(ctx)
+	if !ok {
+		return nil, errors.New("no namespace found")
+	}
+	role, err := c.client.Roles(ns).Get(id)
+	if err != nil {
+		return nil, fmt.Errorf("get role: %v", err)
+	}
+	return role, nil
+}
+
+func (c *rbacClient) ListRoleBindings(ctx api.Context, options *api.ListOptions) (*rbac.RoleBindingList, error) {
+	ns, ok := api.NamespaceFrom(ctx)
+	if !ok {
+		return nil, errors.New("no namespace found")
+	}
+	roleBindings, err := c.client.RoleBindings(ns).List(*options)
+	if err != nil {
+		return nil, fmt.Errorf("list role bindings: %v", err)
+	}
+	return roleBindings, nil
+}
+
+func init() {
+	admission.RegisterPlugin("TaintAdmit", func(client clientset.Interface, config io.Reader) (admission.Interface, error) {
+		return NewTaintAdmit(client, config), nil
+	})
+}
+
+// TaintAdmit is an implementation of admission.Interface which taints nodes at creation time
+type taintAdmit struct {
+	client clientset.Interface
+}
+
+// Determine whether a node is untrusted
+func isUntrusted(node *api.Node) (bool, error) {
+	taints, err := api.GetTaintsFromNodeAnnotations(node.Annotations)
+	if err != nil {
+		return false, err
+	}
+	for _, taint := range taints {
+		if taint.Key == "Untrusted" {
+			return true, nil
+		}
+	}
+	return false, nil
+}
+
+func toleratesUntrusted(pod *api.Pod) (bool, error) {
+	tolerations, err := api.GetTolerationsFromPodAnnotations(pod.Annotations)
+	if err != nil {
+		return false, err
+	}
+	for _, toleration := range tolerations {
+		if toleration.Key == "Untrusted" {
+			return true, nil
+		}
+	}
+	return false, nil
+}
+
+// Flag a node as untrusted
+func invalidateNode(node *api.Node) error {
+	if node.Annotations == nil {
+		node.Annotations = make(map[string]string)
+	}
+	taints, err := api.GetTaintsFromNodeAnnotations(node.Annotations)
+	newTaints := []api.Taint{}
+	untrustedTaint := api.Taint{
+		Key:    "Untrusted",
+		Value:  "true",
+		Effect: api.TaintEffectNoSchedule,
```

Comment:
Same comment as in the tpm manager ... do we want to be more strict / evict existing pods?